netbox/netbox_inventory.py

```diff
@@ -71,57 +80,98 @@ def cli():
+    def prepare_request(self, request: requests.Request) -> requests.PreparedRequest:
+        """Prepare the request by appending the base URL to the request URL.
+
+        Args:
+            request: The request object to prepare.
+
+        Returns:
+            requests.PreparedRequest: The prepared request object.
+        """
         request.url = self.url + request.url
         return super(NetBoxSession, self).prepare_request(request)
 
 
-netbox: NetBoxSession = None
+NETBOX_SESSION: Optional[NetBoxSession] = None
+
 
+@lru_cache
+def get_site(site_slug: str) -> dict:
+    """Retrieve details of a site from NetBox using its slug.
 
-@lru_cache()
-def get_site(site_slug):
-    res = netbox.get("/api/dcim/sites/", params={"slug": site_slug})
+    Args:
+        site_slug: The slug of the site.
+
+    Returns:
+        dict: The site details as a dictionary.
+    """
+    res = NETBOX_SESSION.get("/api/dcim/sites/", params={"slug": site_slug})
     res.raise_for_status()
     return res.json()["results"][0]
 
 
-def create_csv_file(inventory_records, cli_opts):
+def create_csv_file(inventory_records: Iterator[dict], cli_opts: argparse.Namespace) -> None:
+    """Create a CSV file from inventory records.
+
+    Args:
+        inventory_records: An iterator of device records.
+        cli_opts: Parsed command-line options, including the output file.
+    """
     csv_wr = csv.writer(cli_opts.output)
     csv_wr.writerow(CSV_FIELD_NAMES)
 
     for rec in inventory_records:
         csv_wr.writerow(rec_to_csv(rec))
 
 
-def fetch_inventory(cli_opts):
-    global netbox
+def fetch_inventory(cli_opts: argparse.Namespace) -> Iterator[dict]:
+    """Fetch the inventory of devices from NetBox based on the provided CLI options.
+
+    Args:
+        cli_opts: Parsed command-line options.
+
+    Returns:
+        Iterator[dict]: An iterator of filtered device records.
+    """
+    global NETBOX_SESSION
 
     try:
         nb_url = os.environ["NETBOX_ADDR"]
         nb_token = os.environ["NETBOX_TOKEN"]
     except KeyError as exc:
-        sys.exit(f"ERROR: missing envirnoment variable: {exc.args[0]}")
+        sys.exit(f"ERROR: missing environment variable: {exc.args[0]}")
 
-    netbox = NetBoxSession(url=nb_url, token=nb_token)
+    NETBOX_SESSION = NetBoxSession(url=nb_url, token=nb_token)
 
-    # -------------------------------------------------------------------------
-    # perform a GET on the API URL to obtain the Netbox version; the value is
-    # stored in the response header.  convert to tuple(int) for comparison
-    # purposes.  If the Netbox version is after 2.6 the API status/choice
-    # changed from int -> str.
-    # -------------------------------------------------------------------------
-
-    res = netbox.get("/api")
+    # Perform a GET on the API URL to obtain the Netbox version
+    res = NETBOX_SESSION.get("/api")
     api_ver = tuple(map(int, res.headers["API-Version"].split(".")))
```

Comment:
**Suggestion:** Add a check to ensure that the API version header is present in the response to avoid `KeyError` when accessing it. [possible bug, importance: 8]
```suggestion
    api_version = res.headers.get("API-Version")
    if api_version is None:
        raise ValueError("API-Version header is missing in the response")
    api_ver = tuple(map(int, api_version.split(".")))
```